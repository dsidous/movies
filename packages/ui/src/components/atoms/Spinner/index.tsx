import React from 'react';

import useStyles from './styles';

const Spinner = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <i className="fa fa-spinner fa-pulse fa-3x fa-fw" />
      <span className="sr-only">Loading...</span>
    </div>
  );
};

export default Spinner;
