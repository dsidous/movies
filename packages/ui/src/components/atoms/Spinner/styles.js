import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  {
    root: {
      display: 'grid',
      gridTemplateRows: '100vh',
      width: '100%',

      '& i': {
        alignSelf: 'center',
        justifySelf: 'center',
      },
    },
  },
  { name: 'spinner' },
);
