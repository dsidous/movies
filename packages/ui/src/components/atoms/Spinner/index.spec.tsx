import React from 'react';
import { shallow } from 'enzyme';

import Spinner from '.';

describe('Atoms/Spinner', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Spinner />);

    expect(wrapper).toMatchSnapshot();
  });
});
