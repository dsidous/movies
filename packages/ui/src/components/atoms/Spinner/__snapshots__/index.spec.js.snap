// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Spinner should render as expected 1`] = `ShallowWrapper {}`;
